src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1808,6 +1956,105 @@ open_common(int (*real_open)(const char *pathname, int oflags, ...), const char
 	if (!is_target_path)
 		goto org_func;
 
+	/* Always rely on fuse for open() to avoid a fake fd */
+	if (compatible_mode) {
+		dfs_obj_t    *dfs_obj_local;
+		struct ht_fd *fd_ht_obj = NULL;
+		int           fd_fake;
+
+		if (two_args)
+			fd_kernel = real_open(pathname, oflags);
+		else
+			fd_kernel = real_open(pathname, oflags, mode);
+		/* dfuse fails for some reason */
+		if (fd_kernel < 0)
+			goto out_compatible;
+		/* query file object through ioctl() */
+		rc = fetch_dfs_cont_file_obj_with_fd(fd_kernel, dfs_mt, &dfs_obj_local);
+		if (rc)
+			goto out_compatible;
```

Comment:
error ignored?

Reply:
Error message was printed inside fetch_dfs_cont_file_obj_with_fd() already. ok. I can print a debug message about the failure of fetch_dfs_cont_file_obj_with_fd() here.

Reply:
fetch_dfs_cont_file_obj_with_fd() is called inside open. When it is called, we already get fd from dfuse without issue. Even fetch_dfs_cont_file_obj_with_fd fails, we still return the fd allocated by dfuse.